### ISSUE TITLE
./init.sh script adding extra return characters "^M" on daq-site computer

### ISSUE THREAD

**Issue by chw3k5** [simonsobs/TeleView#23]
The init.sh script is supposed to copy a fraction of the variables from the .env file into a the required format for next.js, a file called .evn.production. When this file is made on a mac (bash) or windows (git-bash) it works as expected.

However, on the `daq-site` computer, it add an extra `^M `character to each line, sometimes at the end of the line. Sometimes at the before the last double-quote `"`.  The is some sort of line return character.

A two hours of Google searching did not reveal the cause of this issue. No one else seems to have it.

Is it the version of bash?

Should I just pick a different interpreter with the hashbang at the top of the file?

Did I write the [init.sh](https://github.com/simonsobs/TeleView/blob/main/init.sh) script in wrong way?

Ideally this script would not need to exist. However it seems that some information, specifically, the host name TELEVIEW_PUBLIC_SITE_HOST, is needed at the NEXT.js docker image build-time (see issue #21).

**Comment by mhasself** [comment]
This issue tripped me up when I was trying to set up for debugging. 

These line endings can be changed if the file was at some point edited in Windows.  Look up "Windows vs. Unix line endings".

[This link](https://superuser.com/questions/194668/grep-to-find-files-that-contain-m-windows-carriage-return) has helpful tips for finding and cleaning such files.  I'll PR something so that init.sh checks for this.

**Comment by chw3k5** [comment]
I expected the Windows CLRF to be \r\n, not ^M. I had a hard time with this, I remember losing a day on it before giving up and writing the issue report.

My Windows machine is set to commit or checkout unix style line endings.

I am checking my understand of what must have happened. I (must have) made the .env on Windows, sftp it to the simons-site server. Even though the .env file looked fine in vim, and worked to hydrate the docker compose.yaml file, it still contained an insidious line ending. When I used bash to read and write parts of that file to make the development variables for the next.js application, at that point those line ending resurfaced. I either tested on the production branch or copied the prodcution .env file for testing. So you hit the same issue I reported.

Or did you hit this during testing of a fresh pull from GitHub using the .env-default (`cp .evn-default .env`)? That is what I usually did when I tested on one of my Macs and I never saw the issue. If you copied from .evn-default,  That would be scary as I am expecting git auto change the line endings on commit of .env-default.

**Comment by mhasself** [comment]
> Or did you hit this during testing of a fresh pull from GitHub using the .env-default (`cp .evn-default .env`)?

No, I copied the active `.env` file.  (So in fact, it remains contaminated on the server.)  The github version is clean.
